src/core/Facemesh.tsx

```diff
@@ -41,17 +45,15 @@ export const Facemesh = React.forwardRef<FacemeshApi, FacemeshProps>(
   ) => {
     const outerRef = React.useRef<THREE.Group>(null)
     const meshRef = React.useRef<THREE.Mesh>(null)
-    const [geometry] = React.useState(() => new THREE.BufferGeometry().setIndex(FacemeshDatas.TRIANGULATION))
 
     const [sightDir] = React.useState(new THREE.Vector3())
```

Comment:
This should be a callback.
const [sightDir] = React.useState(() => new THREE.Vector3())

Reply:
@CodyJasonBennett also [already advised](https://discord.com/channels/740090768164651008/740093168770613279/1105616773472059474) me to use what he called a lazy setter, but can you explain why: is this for performance/non-blocking reasons? In my mind, executing `new Vector3()` was instant, isn't it?

Reply:
Yes and it will execute instantly whenever the component renders. Putting it in the callback will make sure it only executes once. :p

Reply:
Maybe I'm wrong, but in my mind, wether `initialState` is an expression or a function:
> This argument is ignored after the initial render.

-- [`useState` doc](https://react.dev/reference/react/useState#parameters:~:text=This%20argument%20is%20ignored%20after%20the%20initial%20render)

Reply:
I’d have to sit down to really test it, but what I can say just looking at the code is that even if it is ignoring the initial value after the first render, you are still executing a function as a parameter each render. So you are eating the cost of creating a new object which then gets ignored. If it is a callback then it just doesn’t call the function in the first place.

Reply:
It gets worst with a Buffer, which will keep allocating memory without any cleanup.

Reply:
ok, I understand that: in effective, the result of `new Vector3()` should be ignored but still executed, so better to put it into a function: `() => new Vector3()` so it is ignore AND not executed ;)

Reply:
thank you @krispya for this explanation